modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupContext.java

```diff
@@ -1204,12 +1207,28 @@ public void onReconnected() {
     }
 
     /**
-     * WAL enabled flag.
+     * Value returned by this method can be changed runtime by the user or during rebalance.
+     *
+     * @return WAL enabled flag.
+     * @see IgniteCluster#disableWal(String)
+     * @see IgniteCluster#enableWal(String)
      */
     public boolean walEnabled() {
```

Comment:
Yes, but it's an out of scope of current ticket.

Reply:
That's in scope, see comment above.
To simplify everything (you added `grp.persistenceEnabled()` everywhere, while it should be encapsulated) after the change we must have 3 methods:
`localWalEnabled`, `walEnabled`, `walOrCdcEnabled` where last two uses the first.

Reply:
> That's in scope, see comment above.

No, it's not.

Reply:
> To simplify everything (you added grp.persistenceEnabled() everywhere, while it should be encapsulated) after the change we must have 3 methods:

Sorry, this can't be done during this ticket.

Reply:
> you added grp.persistenceEnabled() everywhere

Can you, please, clarify what do you mean by "everywhere"?
`persistenceEnabled` added 2 times where it should be to exclude unnecessary records in the WAL for in-memory groups.

Reply:
every `walEnabled ` usage (except CheckpointWorkflow) now covered by existing or added `&& grp.persistenceEnabled()`.
It's good to encapsulate this to `walEnabled ` with renaming of current `walEnabled` to `localWalEnabled`
Just to minimize length of if's conditions.

Reply:
> It's good to encapsulate this to walEnabled with renaming of current walEnabled to localWalEnabled

Refactoring is not the goal of this ticket.

Reply:
The goal is to have good design each time.
 Current case seems to be unfinished. And can be finished in way I described.

Reply:
`walEnabled` method doesn't changed in the current ticket.
I don't see how renaming relates to the current PR.
Let's do renaming in other ticket?